src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -291,6 +291,67 @@ private void BoundsCheck(SequencePosition start, SequencePosition position)
             }
         }
 
+        internal bool TryGetMemoryList(out IMemoryList<T> startSegment, out int startIndex, out IMemoryList<T> endSegment, out int endIndex)
+        {
+            if (Start.GetObject() == null  || GetSequenceType() != SequenceType.MemoryList)
+            {
+                startSegment = null;
+                endSegment = null;
+                startIndex = 0;
+                endIndex = 0;
+                return false;
+            }
+
+            startIndex = GetIndex(Start.GetInteger());
+            endIndex = GetIndex(End.GetInteger());
+            startSegment = (IMemoryList<T>)Start.GetObject();
+            endSegment = (IMemoryList<T>)End.GetObject();
+            return true;
+        }
+
+        internal bool TryGetArray(out ArraySegment<T> array)
+        {
+            if (Start.GetObject() == null  || GetSequenceType() != SequenceType.Array)
```

Comment:
Does it make sense to cache the object into a local instead of calling Start.GetObject() multiple times? Or since it is just a lightweight accessor of the private field, it doesn't matter here.

The same would apply to GetInteger.

Reply:
Implementations are so tiny, they should just get inlined.